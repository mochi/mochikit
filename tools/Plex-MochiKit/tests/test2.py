import Test
from Plex import *
import sys

lex = Lexicon([
  (Seq(Any("ab"), Rep(Any("ab01"))), 'ident'),
  (Seq(Any("01"), Rep(Any("01"))), 'num'),
  (Any(' \n'), IGNORE),
  (Str("abba"), 'abba'),
  (Any('([{!"#') + Rep(AnyBut('!"#}])')) + Any('!"#}])'), IGNORE)
],
debug = Test.debug,
timings = sys.stderr
)

Test.run(lex, "test2", debug = 0, trace = 0)


