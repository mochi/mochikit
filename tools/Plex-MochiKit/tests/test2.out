test2.in   2   0 ident      'a'
test2.in   2   2 ident      'b'
test2.in   2   4 ident      'ab'
test2.in   2   7 ident      'abba'
test2.in   2  12 ident      'a0'
test2.in   3   0 ident      'b1'
test2.in   3   3 ident      'abab0110bba1'
test2.in   4   0 num        '0'
test2.in   4   2 num        '1'
test2.in   4   4 num        '00'
test2.in   4   7 num        '01'
test2.in   5   0 num        '101010'
test2.in   5   7 num        '0001010101'
test2.in   8   0 None       ''
